### ISSUE TITLE
SCSS variable autocomplete gets two `$` prefix instead of one

### ISSUE THREAD

**Issue by mrleblanc101** [mrmlnc/vscode-scss#172]
- VS Code Version: 1.83.1
- SCSS IntelliSense Version: 0.10.0
- Operating System: macOS Ventural 14.1

In a SCSS file, if I autocomplete a variable name, then change the variable, autocomplete work correctly.
![2023-10-30 11 24 08](https://github.com/vuejs/language-tools/assets/17392251/9b27cba3-95b2-49df-8dea-f7fff50509cd)

In a Vue file, if I autocomplete a variable name, then change the variable, the autocomplete will add the `$` prefix even if it's already present leading to a double `$$` before the variable name.
![2023-10-30 11 21 42](https://github.com/vuejs/language-tools/assets/17392251/4674537b-aa90-48d9-8b78-14466204582f)

This seems to be a conflict betweent SCSS IntelliSense (vscode-scss) and Volar.
I opened the issue in the Volar repo too: https://github.com/vuejs/language-tools/issues/3695

**Comment by luqiudi** [comment]
it is super annoying.
how can i solve it

**Comment by joy-yu** [comment]
We can use [vue-scss-variable-scan](https://marketplace.visualstudio.com/items?itemName=Rain.vue-scss-variable-scan) to get the same extension functionality in Vue project.

**Comment by iShawnWang** [comment]
> We can use [vue-scss-variable-scan](https://marketplace.visualstudio.com/items?itemName=Rain.vue-scss-variable-scan) to get the same extension functionality in Vue project.

still double $ for me

**Comment by nofx77** [comment]
Still super annoying 2 years later...

**Comment by Me-Phew** [comment]
@nofx77 

I've added a PR https://github.com/mrmlnc/vscode-scss/pull/178 with a really simple workaround I've made for myself a while ago.
I'm not sure if it is the preferred solution vs https://github.com/mrmlnc/vscode-scss/pull/171 which has a substantial amount of changes.

Also I don't think this repo is really maintained anymore so it is unlikely any of them will ever be merged.
Thus I have uploaded a compiled vsix with changes from https://github.com/mrmlnc/vscode-scss/pull/178 included to https://github.com/Me-Phew/vscode-scss/blob/master/vscode-scss-0.10.0.vsix if anyone wants to use it.

**Comment by mrleblanc101** [comment]
@Me-Phew Thanks, but it doesn't seem to be working all the time.
Since the behavior is inconsistant, then sometime your fix will remove the `$` when it shouldn't

![Image](https://github.com/user-attachments/assets/bc61f788-8aa0-4843-901e-a1c42da87f5a)

**Comment by Me-Phew** [comment]
Yeah now that you mention it I remember why I didn't submit this PR back then. 

I forgot it might only be working in my specific case because I have another extension that always adds the $ for some reason. 

I can't remember which extension it is but I'll let you know if I find out.

**Comment by mrleblanc101** [comment]
@Me-Phew The original bug is inconsistent. 

![Image](https://github.com/user-attachments/assets/bc82a670-995e-4827-aa7f-fd0e764ddc29)

So your fix is inconsistent.
I don't thinks it's related to another extension (other than the Vue extension).
I don't know how VS Code extension works, but maybe you can add a check and only remove the `$` if there is already one instead of removing it all the time.

I'll try #171

**Comment by mrleblanc101** [comment]
@Me-Phew I manually compiled #171 and it doesn't work at all.
It removes the `$` completely, but if you add it yourself there is no suggestions at all.

![Image](https://github.com/user-attachments/assets/43b65628-18fd-46dd-8732-6a6ea32ed2cd)
